bigtable-hbase-2.x-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/HbaseToCloudBigtableReplicationEndpoint.java

```diff
@@ -0,0 +1,67 @@
+package com.google.cloud.bigtable.hbase.replication;
+
+import com.google.cloud.bigtable.hbase.replication.adapters.BigtableWALEntry;
+import org.apache.hadoop.hbase.replication.BaseReplicationEndpoint;
+import org.apache.hadoop.hbase.wal.WAL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+// TODO(remove BaseReplicationEndpoint extension).
+public class HbaseToCloudBigtableReplicationEndpoint extends BaseReplicationEndpoint {
+    private static final Logger LOG =
+        LoggerFactory.getLogger(HbaseToCloudBigtableReplicationEndpoint.class);
+
+    private final CloudBigtableReplicator cloudBigtableReplicator;
+    public HbaseToCloudBigtableReplicationEndpoint() {
+        super();
+        cloudBigtableReplicator = new CloudBigtableReplicator();
+    }
+
+    @Override
+    public UUID getPeerUUID() {
+        return cloudBigtableReplicator.getPeerUUID(); }
+
+    @Override
+    public boolean replicate(ReplicateContext replicateContext) {
+        Map<String, List<BigtableWALEntry>> walEntriesByTable = new HashMap<>();
+        for (WAL.Entry wal: replicateContext.getEntries()) {
+            String tableName = wal.getKey().getTableName().getNameAsString();
+            BigtableWALEntryImpl bigtableWALEntryImpl = new BigtableWALEntryImpl(wal);
```

Comment:
Here, you can just call the core BigtableEntry constrcutor, you won't need the BigtableEntryImpl class.